src/main/java/de/hysky/skyblocker/skyblock/chat/SkyblockXpMessages.java

```diff
@@ -0,0 +1,37 @@
+package de.hysky.skyblocker.skyblock.chat;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import de.hysky.skyblocker.config.SkyblockerConfigManager;
+import de.hysky.skyblocker.utils.Constants;
+import de.hysky.skyblocker.utils.Utils;
+import de.hysky.skyblocker.utils.scheduler.Scheduler;
+import it.unimi.dsi.fastutil.ints.IntOpenHashSet;
+import net.fabricmc.fabric.api.client.message.v1.ClientReceiveMessageEvents;
+import net.minecraft.client.MinecraftClient;
+import net.minecraft.text.Text;
+
+public class SkyblockXpMessages {
+	private static final MinecraftClient CLIENT = MinecraftClient.getInstance();
+	private static final Pattern SKYBLOCK_XP_PATTERN = Pattern.compile("§b\\+\\d+ SkyBlock XP §7\\([^()]+§7\\)§b \\(\\d+\\/\\d+\\)");
+	private static final IntOpenHashSet RECENT_MESSAGES = new IntOpenHashSet();
+
+	public static void init() {
+		ClientReceiveMessageEvents.GAME.register(SkyblockXpMessages::onMessage);
+		Scheduler.INSTANCE.scheduleCyclic(RECENT_MESSAGES::clear, 20 * 60);
```

Comment:
technically possible for the mod to send the message twice when done like this

Reply:
should I just eliminate the cache all together?